### PR TITLE
[flutter_tool] partial null safety migration of tool source code

### PR DESCRIPTION
**Pull request by jonahwilliams**
Migrates all lib/ code to null safety for the flutter tool. This requires ignoring a warning on the import of package:dwds, which is not null safe yet.

Additionally tests were not migrated, because dart migrate was having problems with them. We may need a more incremental approach there.

https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

File: packages/flutter_tools/lib/runner.dart [hunk 1]
```diff
@@ -82,17 +82,17 @@ Future<int> run(
...
 
 Future<int> _handleToolError(
-  dynamic error,
-  StackTrace stackTrace,
+  Object error,
+  StackTrace? stackTrace,
   bool verbose,
   List<String> args,
-  bool reportCrashes,
+  bool? reportCrashes,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/runner.dart:95]: seems like this should be non-nullable (or optional with a default)? We just throw a bang on it later

Comment (jonahwilliams) [reply packages/flutter_tools/lib/runner.dart]: fixed

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart [hunk 2]
```diff
@@ -49,7 +47,7 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
 
   static const Duration launchInBrowserTimeout = Duration(seconds: 15);
 
-  final DevtoolsLauncher _devToolsLauncher;
+  final DevtoolsLauncher? _devToolsLauncher;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/resident_devtools_handler.dart:50]: would it be valid for this to be null?

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart [hunk 3]
```diff
@@ -71,34 +69,34 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
   // This must be guaranteed not to return a Future that fails.
   @override
   Future<void> serveAndAnnounceDevTools({
-    Uri devToolsServerAddress,
-    @required List<FlutterDevice> flutterDevices,
+    Uri? devToolsServerAddress,
+    required List<FlutterDevice?> flutterDevices,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/resident_devtools_handler.dart:73]: Is it valid to have nulls in this list (and those in the other methods here)?

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:50]: we have tests that assert that the flutter daemon works when this is null

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:73]: See https://github.com/flutter/flutter/pull/105798/files/d337ee3f6ded659db096998343f7772f41bf8289#diff-2fb6523413045c1c558c5c124f3ec97d6f307079f129584d81d02ba59486ef0cR174

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:73]: hmm that didn't work

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:73]: look at waitForExtensionForDevice, it returns `null` for the FlutterDevice if it fails

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:73]: good to know. thanks

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart:50]: :+1:

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 4]
```diff
@@ -178,15 +176,15 @@ class Daemon {
...
   final DaemonConnection connection;
 
-  DaemonDomain daemonDomain;
-  AppDomain appDomain;
-  DeviceDomain deviceDomain;
-  EmulatorDomain emulatorDomain;
-  DevToolsDomain devToolsDomain;
-  ProxyDomain proxyDomain;
-  StreamSubscription<DaemonMessage> _commandSubscription;
+  late DaemonDomain daemonDomain;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/daemon.dart:179]: this looks sus, but i doubt we can fix it without a refactor.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
of course removing the meta dep breaks g3

**Pull request merged**